base/reducedim.jl

```diff
@@ -304,6 +304,22 @@ sum(A, dims)
     sum!(r, A)
 
 Sum elements of `A` over the singleton dimensions of `r`, and write results to `r`.
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+ 
+julia> sum!([1; 1], A)
+2-element Array{Int64,1}:
```

Comment:
Would something like:

```julia
julia> r = [1;1]; sum!(r, A); r
2-element Array{Int64,1}:
 3
 7
```

to show that `r` get modified be useful? Perhaps it is just cluttering things.